src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -333,33 +336,66 @@ private IList<string> GetConfigFilePaths(ISettings settings)
             return settings.GetConfigFilePaths();
         }
 
-        private static PackageReference[] GetPackageReferences(PackageSpec packageSpec)
+        private async Task<IEnumerable<PackageReference>> GetPackageReferences(PackageSpec packageSpec)
         {
             var frameworkSorter = new NuGetFrameworkSorter();
 
+            var lockFilePath = await GetAssetsFilePathAsync();
```

Comment:
naming: Call it the assets file path. The type is called lock file and we can't change that because of backwards compat, but we should use assets file wherever possible.